inference-engine/src/vpu/graph_transformer/src/stages/interpolate.cpp

```diff
@@ -25,92 +27,89 @@ void FrontEnd::parseInterpolate(const Model& model, const ie::CNNLayerPtr& _laye
-                                                       _layer->name,
-                                                       _layer,
-                                                       antialias,
-                                                       coordinateTransformationMode,
-                                                       nearestMode,
-                                                       -1.0f,
-                                                       input,
-                                                       output);
-            } else if (cmp(interpolateMode, "linear") || cmp(interpolateMode, "linear_onnx")) {
-                // current "Interp" supports modes "align_corners" and "asymmetric" only
-                // other "Interpolate" modes are translated to the default ones
-                const auto coordinateTransformationMode = _layer->GetParamAsString("coordinate_transformation_mode", "half_pixel");
-
-                _stageBuilder->addInterpStage(model,
-                                              _layer->name,
-                                              _layer,
-                                              cmp(coordinateTransformationMode, "align_corners"),
-                                              input,
-                                              output);
-            } else {
-                VPU_THROW_EXCEPTION << "Current Interpolate supports 'nearest' and 'linear' modes only; layer name = " << _layer->name;
-            }
-        } else {
-            VPU_THROW_EXCEPTION << "Current Interpolate does not support paddings, batches, and resize by channels; layer name = " << _layer->name;
-        }
-    } else {
-        VPU_THROW_EXCEPTION << "Current Interpolate supports (N)HWC, (N)CHW data orders only; layer name = " << _layer->name;
+
+    const auto ic = input->desc().dim(Dim::C);
+    const auto in = (input->desc().numDims() == 3) ? 1 : input->desc().dim(Dim::N);
+
+    const auto oc = output->desc().dim(Dim::C);
+    const auto on = (output->desc().numDims() == 3) ? 1 : output->desc().dim(Dim::N);
+
+    auto padsBegin = _layer->GetParamAsInts(g_pads_begin, {});
+    auto padsEnd   = _layer->GetParamAsInts(g_pads_end, {});
+
+    const auto isPadZeros = [](const std::vector<int>& pad) {
+        return std::all_of(pad.begin(), pad.end(), [](int i) { return i == 0; });
+    };
+
+    ie::details::CaselessEq<std::string> cmp;
+    const auto orderIsSupported = input->desc().dimsOrder() == DimsOrder::NCHW || input->desc().dimsOrder() == DimsOrder::NHWC
+                               || input->desc().dimsOrder() == DimsOrder::CHW ||  input->desc().dimsOrder() == DimsOrder::HWC;
+    VPU_THROW_UNLESS(orderIsSupported, "Current Interpolate supports (N)HWC, (N)CHW data orders only, actual {}", input->desc().dimsOrder());
+
+    const auto modeIsSupported = cmp(interpolateMode, g_nearest) || cmp(interpolateMode, g_linear) || cmp(interpolateMode, g_linear_onnx);
```

Comment:
The same: you can use interpModeMap.find and then use only enum values for conditions. In such a case you can eliminate using `cmp` and string variables